### ISSUE TITLE
Connection error

### ISSUE THREAD

**Issue by brennenawana** [ewrogers/gala-docker#5]
I am running a fair amount of nodes and once in a while I'll notice that not all of them are running. When looking at the logs I see the following message:
`error "caught err in main: Error: Client network socket disconnected before secure TLS connection was established"`

Is it possible for this to self-heal or restart when this happens?

**Comment by ewrogers** [comment]
Yes, we could use a `HEALTHCHECK CMD ...` script to monitor the progress and have it auto-restart if something isn't right.

I can look into this, but won't be able to add such a feature until next week.
